jck/update_git_material.sh

```diff
@@ -0,0 +1,151 @@
+    mkdir -p "$(dirname "$clone_directory")"
+    # Check if the parent directory was created successfully
+    if [ $? -eq 0 ]; then
+        # Change into the parent directory
+        cd "$(dirname "$clone_directory")" || exit 1
+        # Clone the repository into the target directory
+        git clone --single-branch -b "${jck_branch}" "${JCK_GIT_REPO_USED}" "$(basename "$clone_directory")"
+        # Check if the clone was successful
+        if [ $? -eq 0 ]; then
+            echo "Repository cloned successfully into $clone_directory"
+        else
+            echo "Failed to clone repository"
+            exit 1
+        fi
+    else
+        echo "Failed to create parent directory: $(dirname "$clone_directory")"
+        exit 1
+    fi
+}
+
+# Function to remove test material and git clone
+remove_and_clone() {
+    echo "Removing test material and git clone again"
+    rm -rf "$JCK_ROOT_USED"
+    git_clone
+}
+
+# Function to perform hard reset and pull
+perform_hard_reset_and_pull() {
+    echo "Performing hard reset..."
+    git reset --hard "origin/$jck_branch" || {
+        echo "Hard reset failed. Removing directory and cloning again."
+        remove_and_clone
+    }
+
+    # Clean untracked files and directories
+    git clean -dfx || {
+        echo "Failed to clean untracked files and directories."
+        remove_and_clone
+    }
+
+    # Pull the latest changes
+    git pull "$JCK_GIT_REPO_USED" "$jck_branch" || {
+        echo "Pull failed. Removing directory and cloning again."
+        remove_and_clone
+    }
+}
+
+# Function to check for updates
+check_and_handle_updates() {
+    local localSHA=$(git rev-parse HEAD)
+    localSHATrimmed=$(echo "${localSHA}" | tr -d '[:space:]')
+    local remoteSHA=$(git ls-remote "$JCK_GIT_REPO_USED" "$jck_branch" | cut -f1)
+
+    echo "LocalSHA = --$localSHA--"
+    echo "RemoteSHA = --$remoteSHA--"
+
+    # Check if SHA match
+    if [ "$localSHA" != "$remoteSHA" ]; then
+        # Update materials
+        delete_native_dirs
+        echo message="Updating ${JCK_ROOT_USED} with latest..."
+        cd "${JCK_ROOT_USED}" || exit
+        git pull "$JCK_GIT_REPO_USED" "$jck_branch" || {
+            perform_hard_reset_and_pull
+        }         
+    else
+        # Set the property indicating that local materials are up-to-date
+         echo "Local JCK materials up-to-date. Skipping update"
+        local_material_uptodate="true"
+    fi
+
+    # If SHA match or update succeeded
+   echo "Local JCK materials updated"
+
+    # Check if isZOS property is set
+    if [ -n "$isZOS" ] && [ "$isZOS" = "true" ]; then
+        # Check if local-material-uptodate property is set
+        if [ -n "$local_material_uptodate" ]; then
+            echo "zOS Local JCK materials up-to-date. Skipping hard reset"
+        else
+            # Perform hard reset for z/OS
+            perform_hard_reset_zos
+        fi
+    fi
+}
+
+# Function to perform hard reset for z/OS
```

Comment:
Why is this needed?

Reply:
Will need to remove this function as `perform_hard_reset_and_pull` needs to handle both ZOS and non-ZOS.